cla_backend/apps/legalaid/migrations/0020_fill_missing_outcome_codes.py

```diff
@@ -0,0 +1,38 @@
+# coding=utf-8
+from __future__ import unicode_literals
+import logging
+
+from django.db import models, migrations
+
+logger = logging.getLogger(__name__)
+
+
+def re_denormalize_outcome_codes_to_cases(apps, schema_editor):
+    try:
+        from cla_eventlog.constants import LOG_LEVELS, LOG_TYPES
+        from cla_eventlog.models import Log
+    except ImportError:
+        pass
```

Comment:
If this exception happens, would the migration be considered successful? Would there be a way to retry?
Is it safe to do data fixes during migrations?

Reply:
It would be considered successful if the exception occurred. This migration isn't critical and the failure is explicitly caught by the monitoring pieces. Anyhow, migration dependencies updated and historical version of the Log model is now referred to obviating the error handling.

Yes, [common practice](https://docs.djangoproject.com/en/1.7/topics/migrations/#data-migrations)
 in Django.